tests/testthat/test_intervals.R

```diff
@@ -145,3 +146,61 @@ test_that("overlapping age intervals are correctly identified", {
   expect_identical(output, expected)
 })
 
+# Test that intervals are collapsed correctly to common set ---------------
+
+id_cols <- c("year_start", "year_end", "sex", "age_start", "age_end")
+value_cols <- c("value")
+
+# set up test input data.table
```

Comment:
Perhaps test case where intervals in one case can't directly be aggregated to intervals in another case. Example: [0, 5, 10, 20] & [0, 5, 15, 20] -- > [0, 5, 20].

Reply:
Thanks for suggesting this! My most recent commit adds a test for this and fixes to make it work